packages/app/src/app/hooks/useGitHubAuthorization.ts

```diff
@@ -0,0 +1,30 @@
+import { useAppState } from 'app/overmind';
+
+const NO_AUTH = {
+  allowsPrivateRepos: false,
+  allowsPublicRepos: false,
+};
+
+export const useGitHubAuthorization = (): {
+  allowsPrivateRepos: boolean;
+  allowsPublicRepos: boolean;
+} => {
+  const { hasLogIn, user } = useAppState();
+
+  if (!hasLogIn || !user) {
+    return NO_AUTH;
+  }
+
+  const { data, error } = user.githubProfile;
+  if (error) {
+    return NO_AUTH;
+  }
+
+  const scopes = data.scopes;
+
+  return {
+    allowsPrivateRepos: scopes.includes('repo'),
+    allowsPublicRepos:
+      scopes.includes('repo') || scopes.includes('public_repo'),
+  };
+};
```

Comment:
@tristandubbeld do you think it makes sense to have these fields? Or should we aim for a simpler return type?

Reply:
I really like the explicitness @olarclara. I think it's way better than typing it as a string (or number as we do with privacy right now 😬). Do we want to include something like `hasGithubAuthorization` too to maybe check if the user has any authorization at all?

Reply:
@tristandubbeld I spent some time debating about this, at first I was going to return `null` if there was no authorization but I preferred to follow the pattern of `useWorkspaceSubscription` and return sensible defaults. Wdyt?

Reply:
I think that's great!